ssm/agents.py

```diff
@@ -250,6 +250,7 @@ def run_sender(protocol, brokers, project, token, cp, log):
         print('SSM failed to complete successfully.  See log file for details.')
         log.error('Unexpected exception in SSM: %s', e)
         log.error('Exception type: %s', e.__class__)
+        sys.exit(1)
```

Comment:
Putting the exit here will mean lines 255 to 262 don't get executed. I think what we need to do is set a flag here and then use it to set the exit status at the end of `run_sender` (and equivalent for `run_receiver`)